src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -217,6 +217,277 @@ .maxstack 8
+    //         System.Console.WriteLine("In C..ctor before assignment: " + P);
+    Diagnostic(ErrorCode.WRN_UseDefViolationThisSupportedVersion, "P").WithLocation(12, 69)
+    );
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetMembers().OfType<FieldSymbol>());
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void TestPropertyIsAssignedInStructConstructor_PropertyHasSetter()
+        {
+            var comp = CreateCompilation(@"
+public struct C
+{
+    public static void Main()
+    {
+        var x = new C();
+        System.Console.Write(x.P + "" ""); // 5
+        x.P = 10;
+        System.Console.Write(x.P + "" ""); // 10
+        x = new C();
+        System.Console.Write(x.P); // 5
+    }
+
+    public C()
+    {
+        P = 5;
+    }
+
+    public int P { get => field; set => field = value; }
+}
+", options: TestOptions.ReleaseExe.WithSpecificDiagnosticOptions(ReportStructInitializationWarnings));
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            CompileAndVerify(comp, expectedOutput: "5 10 5").VerifyIL("C..ctor", @"
+{
+  // Code size       15 (0xf)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.0
+  IL_0002:  stfld      ""int C.<P>k__BackingField""
+  IL_0007:  ldarg.0
+  IL_0008:  ldc.i4.5
+  IL_0009:  call       ""void C.P.set""
+  IL_000e:  ret
+}
+").VerifyDiagnostics(
+    // (16,9): warning CS9020: The 'this' object is read before all of its fields have been assigned, causing preceding implicit assignments of 'default' to non-explicitly assigned fields.
+    //         P = 5;
+    Diagnostic(ErrorCode.WRN_UseDefViolationThisSupportedVersion, "P").WithLocation(16, 9)
+    );
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetMembers().OfType<FieldSymbol>());
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void TestPropertyIsAssignedInStructConstructor_ReadOnlyProperty()
+        {
+            var comp = CreateCompilation(@"
+public struct C
+{
+    public static void Main()
+    {
+        var x = new C();
+        System.Console.Write(x.P);
+    }
+
+    public C()
+    {
+        P = 5;
+    }
+
+    public int P { get => field; }
+}
+", options: TestOptions.ReleaseExe.WithSpecificDiagnosticOptions(ReportStructInitializationWarnings));
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            CompileAndVerify(comp, expectedOutput: "5").VerifyIL("C..ctor", @"
+{
+  // Code size       15 (0xf)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.0
+  IL_0002:  stfld      ""int C.<P>k__BackingField""
```

Comment:
>stfld      ""int C.<P>k__BackingField"" [](http://example.com/codeflow?start=12&length=39)

Zero initialization looks unnecessary in this case, the field is assigned directly. This looks directly related to the goal of this PR. That assignment in the constructor should be treated as direct field write for the purpose of flow analysis.

Reply:
Note that IL for a regular auto property looks as expected:
```
public struct C
{
    public C()
    {
        P = 5;
    }

    public int P { get; }
}
```

```
.method public hidebysig specialname rtspecialname 
        instance void  .ctor() cil managed
{
  // Code size       9 (0x9)
  .maxstack  8
  IL_0000:  nop
  IL_0001:  ldarg.0
  IL_0002:  ldc.i4.5
  IL_0003:  stfld      int32 C::'<P>k__BackingField'
  IL_0008:  ret
} // end of method C::.ctor
```

So, the problem looks specific to the `field` keyword  feature.

Reply:
Yes it's specific to `field`. Dataflow is using `.BackingField`. It should rely on both `BackingField` and `FieldKeywordBackingField`. I'll attempt to fix it in this PR and see if it's small enough.

Reply:
Fixed in last commit (hope I did it correctly)